src/engine/resource_manager/mod.rs

```diff
@@ -30,28 +31,92 @@ use notify::DebouncedEvent;
 use std::{path::Path, sync::Arc};
 
 pub mod container;
-mod loader;
+pub mod loader;
 pub mod options;
 mod task;
 pub mod watcher;
 
 /// Storage of resource containers.
 pub struct ContainersStorage {
     /// Container for texture resources.
-    pub textures: ResourceContainer<Texture, TextureImportOptions, TextureLoader>,
+    pub textures: ResourceContainer<Texture, TextureImportOptions>,
 
     /// Container for model resources.
-    pub models: ResourceContainer<Model, ModelImportOptions, ModelLoader>,
+    pub models: ResourceContainer<Model, ModelImportOptions>,
 
     /// Container for sound buffer resources.
-    pub sound_buffers:
-        ResourceContainer<SoundBufferResource, SoundBufferImportOptions, SoundBufferLoader>,
+    pub sound_buffers: ResourceContainer<SoundBufferResource, SoundBufferImportOptions>,
 
     /// Container for shader resources.
-    pub shaders: ResourceContainer<Shader, ShaderImportOptions, ShaderLoader>,
+    pub shaders: ResourceContainer<Shader, ShaderImportOptions>,
 
     /// Container for curve resources.
-    pub curves: ResourceContainer<CurveResource, CurveImportOptions, CurveLoader>,
+    pub curves: ResourceContainer<CurveResource, CurveImportOptions>,
+}
+
+pub struct ResourceManagerBuilder {
+    texture_loader: Box<dyn ResourceLoader<Texture, TextureImportOptions>>,
+    model_loader: Box<dyn ResourceLoader<Model, ModelImportOptions>>,
+    sound_buffer_loader: Box<dyn ResourceLoader<SoundBufferResource, SoundBufferImportOptions>>,
+    shader_loader: Box<dyn ResourceLoader<Shader, ShaderImportOptions>>,
+    curve_loader: Box<dyn ResourceLoader<CurveResource, CurveImportOptions>>,
+}
+
+impl ResourceManagerBuilder {
```

Comment:
Builder should have `build` method that returns `ResourceManager`.

Reply:
There was no need for the builder after the change as engine requires a ResourceManager (and not a builder)